ote/src/cljs/ote/views/user_edit.cljs

```diff
@@ -0,0 +1,84 @@
+    form-data))
+
+(defn edit-user
+  "Edit own user info"
+  [e! app]
+  (r/create-class
+    {:reagent-render
+     (fn
+       [e! {:keys [user user-edit params] :as app}]
+       (let [{:keys [email-taken username-taken form-data email-confirmed?]} user-edit]
+         [:div.user-edit.col-xs-12.col-sm-8.col-md-8.col-lg-6
+          [list-header/header app (tr [:common-texts :user-menu-profile])]
+          [form/form
+           {:update! #(e! (uec/->UpdateForm %))
+            :name->label (tr-key [:register :fields])
+            :footer-fn (fn [data]
+                         [:div {:style {:margin-top "1em"}}
+                          [buttons/save {:on-click #(e! (uec/->SaveUserInfo
+                                                          (merge-user-data
+                                                            user
+                                                            (form/without-form-metadata data))
+                                                          (:id params)))
+                                         :disabled (form/disable-save? data)}
+                           (tr [:buttons :save])]
+                          [buttons/cancel {:on-click #(e! (uec/->CancelEdit))}
+                           (tr [:buttons :cancel])]])}
+           [(form/group
+              {:expandable? false :columns 3 :layout :raw :card? false}
+
+              {:name :username :type :string :required? true :full-width? true
+               :placeholder (tr [:register :placeholder :username])
+               :validate [(fn [data _]
+                            (if (< (count data) 3)
+                              (tr [:common-texts :required-field])
+                              (when (not (user/username-valid? data))
+                                (tr [:register :errors :username-invalid]))))
+                          (fn [data _]
+                            (when (and username-taken (username-taken data))
+                              (tr [:register :errors :username-taken])))]
+               :should-update-check form/always-update}
+
+              {:name :name :type :string :required? true :full-width? true
+               :placeholder (tr [:register :placeholder :name])
+               :should-update-check form/always-update}
+              {:name :email :type :string :autocomplete "email" :required? true
+               :full-width? true :placeholder (tr [:register :placeholder :email])
+               :validate [(fn [data _]
+                            (when (not (user/email-valid? data))
+                              (tr [:common-texts :required-field])))
+                          (fn [data _]
+                            (when (and email-taken (email-taken data))
+                              (tr [:register :errors :email-taken])))]
+               :should-update-check form/always-update}
+              (if email-confirmed?
+                (form/info "Käyttäjä on varmentanut sähköpostiosoitteen.")
```

Comment:
localization

Reply:
Yes it's only for admins so no localizations needed.